nest/src/token/pUSD.sol

```diff
@@ -6,137 +6,493 @@ import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/I
+    //using Math for uint256;
+    using FixedPointMathLib for uint256;
 
-    // ========== ROLES ==========
-    bytes32 public constant UPGRADER_ROLE = keccak256("UPGRADER_ROLE");
-    bytes32 public constant MINTER_ROLE = keccak256("MINTER_ROLE");
-    bytes32 public constant BURNER_ROLE = keccak256("BURNER_ROLE");
-    bytes32 public constant VAULT_ADMIN_ROLE = keccak256("VAULT_ADMIN_ROLE");
-    bytes32 public constant PAUSER_ROLE = keccak256("PAUSER_ROLE");
+    // ========== ERRORS ==========
+    error ZeroAddress();
+
+    error InvalidAsset();
+    error InvalidReceiver();
+    error InvalidSender();
+    error InvalidController();
+    error InvalidVault();
+
+    error AssetNotSupported();
+    error TellerPaused();
+
+    // ========== STORAGE ==========
+
+    struct BoringVault {
+        ITeller teller;
+        IVault vault;
+        IAtomicQueue atomicQueue;
+        ILens lens;
+        IAccountantWithRateProviders accountant;
+    }
+
+    /// @custom:storage-location erc7201:plume.storage.pUSD
+    struct pUSDStorage {
+        BoringVault boringVault;
+        uint8 tokenDecimals;
+        string tokenName;
+        string tokenSymbol;
+        uint256 version;
+        IERC20 usdc;
+        IERC20 usdt;
+    }
 
-    // ========== STATE VARIABLES ==========
-    IVault public vault;
-    bool public paused;
+    // keccak256(abi.encode(uint256(keccak256("plume.storage.pUSD")) - 1)) & ~bytes32(uint256(0xff))
+    bytes32 private constant PUSD_STORAGE_LOCATION = 0x54ae4f9578cdf7faaee986bff2a08b358f01b852b4da3af4f67309dae312ee00;
+
+    function _getpUSDStorage() private pure returns (pUSDStorage storage $) {
+        bytes32 position = PUSD_STORAGE_LOCATION;
+        assembly {
+            $.slot := position
+        }
+    }
 
     // ========== EVENTS ==========
     event VaultChanged(address oldVault, address newVault);
-    event Paused(address account);
-    event Unpaused(address account);
+    event Reinitialized(uint256 version);
 
-    // ========== MODIFIERS ==========
-    modifier whenNotPaused() {
-        require(!paused, "PUSD: paused");
-        _;
-    }
+    // ========== ROLES ==========
+    bytes32 public constant VAULT_ADMIN_ROLE = keccak256("VAULT_ADMIN_ROLE");
 
-    // ========== CONSTRUCTOR & INITIALIZER ==========
-    /// @custom:oz-upgrades-unsafe-allow constructor
+    /**
+     * @notice Prevent the implementation contract from being initialized or reinitialized
+     * @custom:oz-upgrades-unsafe-allow constructor
+     */
     constructor() {
         _disableInitializers();
     }
 
-    function initialize(address _vault, address admin) external initializer {
-        __ERC20_init("", ""); // Empty strings since we override name() and symbol()
-        __AccessControl_init();
+    /**
+     * @notice Initialize pUSD
+     * @param owner Address of the owner of pUSD
+     * @param usdc_ Address of the underlying asset
+     * @param usdt_ Address of the underlying asset
```

Comment:
We only need one asset - USDC is the only asset we'll deposit and redeem from pUSD from Nest. Let's remove "USDT" from the code entirely and rename "USDC" to just "asset"